integration_tests/src/main/python/delta_lake_test.py

```diff
@@ -160,3 +160,22 @@ def convert_and_setup_name_mapping(spark):
     with_cpu_session(setup_parquet_table, {"spark.sql.parquet.fieldId.write.enabled": str(enable_deletion_vectors).lower()})
     with_cpu_session(convert_and_setup_name_mapping, conf={"spark.databricks.delta.properties.defaults.enableDeletionVectors": "false"})
     assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.read.format("delta").load(data_path))
+
+@allow_non_gpu(*delta_meta_allow)
+@delta_lake
+@ignore_order(local=True)
+@pytest.mark.skipif(not is_spark_340_or_later(), reason="Deletion Vectors only supported on Spark 3.4.0+")
+def test_delta_filter_out_metadata_col(spark_tmp_path):
+    data_path = spark_tmp_path + "/DELTA_DATA"
+
+    def create_delta(spark):
+        two_col_df(spark, int_gen, int_gen).coalesce(1).write.format("delta") \
+            .option("delta.enableDeletionVectors", "true") \
+            .partitionBy("a").save(data_path)
+
+        spark.sql(f"DELETE FROM delta.`{data_path}` WHERE b = 0")
```

Comment:
Should we assert that we deleted at least one row by looking at the result df returned by this statement?